### PR TITLE
Lookbook documentation for SubHeader component

### PR DESCRIPTION
**Pull request by HDinger**
https://community.openproject.org/projects/openproject/work_packages/55465/activity

### CODE REVIEW

File: lookbook/docs/patterns/11-sub-header.md.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- **Desktop:** A basic [text input](https://primer.style/components/text-input) with a [search icon](https://primer.style/components/text-input#with-leading-and-trailing-visuals) which is mostly used on index pages to filter the elements below.
+- **Mobile:**
+  - There will be an IconButton shown instead of the input
+  - Clicking that, will show the input again and hide all other elements of the SubHeader
+  - An additional cancel button is shown to return to the previous state
+
+**Filter button:** (Optional)A button which is shown next to the filter input (or on the very left, if there is no filter input). Depending on what you need, it is either:
+
+- A normal button
+- A custom component, (e.g our core [FilterButtonComponent](../../inspect/open_project/filter/filter_button/default))
```
Comment (marcalcobe) [inline lookbook/docs/patterns/11-sub-header.md.erb]: @HDinger what is the case for a custom component here in the filter?

File: lookbook/docs/patterns/11-sub-header.md.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+**Text:** (Optional) A bold text shown in the middle of the SubHeader
+
+**Action buttons**: (Optional) A set of buttons that are shown on the right side
+
+
+## Best practices
+
+**Do**
+
+- Limit the actions of the SubHeader to the content specific actions of the page (e. for filtering, or creating a new object).
```
Comment (marcalcobe) [inline lookbook/docs/patterns/11-sub-header.md.erb]: @HDinger I would add to the page content to differentiate it from the actions in the PageHeader which are view/query related

File: lookbook/docs/patterns/11-sub-header.md.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- Use Icon buttons instead of buttons for actions only when the icon is clear enough to describe the action.
+
+**Don't**
+
+
```
Comment (marcalcobe) [inline lookbook/docs/patterns/11-sub-header.md.erb]: Would add the don't:

- Use the SubHeader to perform generic view actions

**Review by marcalcobe — COMMENTED** [review]
@HDinger I added some comments

File: lookbook/docs/patterns/11-sub-header.md.erb (hunk 1 above)
Comment (HDinger) [reply lookbook/docs/patterns/11-sub-header.md.erb]: That is rather technical, but while thinking about it, I guess this should not be part of this documentation, but rather the preview examples. I will change that.

**Pull request merged**